core-lib/demos/applications/InstantMessengerApplication.ns

```diff
@@ -0,0 +1,165 @@
+class InstantMessengerApplication usingPlatform: platform = Value (
+| private actors = platform actors.
+  private Array = platform kernel Array.
+  private TransferArray = platform kernel TransferArray.
+  private Dictionary = platform collections Dictionary.
+  private TransferObject = platform kernel TransferObject.
+|
+)(
+
+public class InstantMessenger new: name  total: size = (
+  | private buddyMap ::= Dictionary new: size.
+    private name = name.
+    private textMessage ::= nil.
+  |
+)(
+
+    public startChat: remoteMessenger = (
+      | pDiscover pName pSend msg pp array |
+
+        pp:: actors createPromisePair.
+
+        pDiscover:: self addMessenger: remoteMessenger. (* returns a far reference of the remote messenger *)
+
+        pName:: pDiscover <-: getName.
+        pName whenResolved: [:remoteName |
+                msg:: ('Hello ' + remoteName).
+                pSend:: sendMessage: remoteName contentMsg: msg.
+
+                pSend whenResolved:[:result |
+                    (result = 'ok')
+                    ifTrue:[
+                         array:: TransferArray new: 2. (* array with the msg to display in the sender actor screen *)
```

Comment:
You got a `TextMessage` class here, which seems to fulfill the same purpose. Would it make sense to use it instead?

Reply:
Fixed